flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBMapState.java

```diff
@@ -595,6 +595,8 @@ private void loadCache() {
 				 */
 				if (lastEntry != null && !lastEntry.deleted) {
 					iterator.next();
+					cacheEntries.add(lastEntry);
+					cacheIndex = 1;
```

Comment:
Thanks for your comments, I quite agree with your comments except that I prefer to call the `#nextEntry()` returned entry as `currentEntry`, since this entry points to the current position before any other `#nextEntry()` is invoked.